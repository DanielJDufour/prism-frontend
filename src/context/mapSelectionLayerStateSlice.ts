import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import type { RootState } from './store';

// This is used by the selection layer to provide the ability to select layers with the mouse.

interface MapSelectionState {
  // Current cells which are selected on the map (holds their code from admin_boundaries.json)
  selectedBoundaries: string[];
  // If true, clicking the map selects a boundary. If set to false, selectedBoundaries is cleared and selection is disabled.
  isSelectionMode: boolean;
}

const initialState: MapSelectionState = {
  selectedBoundaries: [],
  // TODO default false
  isSelectionMode: true,
};

export const mapSelectionLayerStateSlice = createSlice({
  name: 'mapSelectionLayerState',
  initialState,
  reducers: {
    toggleSelectedBoundary: (
      { selectedBoundaries, ...rest },
      { payload }: PayloadAction<string>,
    ) => {
      if (selectedBoundaries.includes(payload)) {
        return {
          ...rest,
          selectedBoundaries: selectedBoundaries.filter(
            code => code !== payload,
          ),
        };
      }
      return {
        ...rest,
        selectedBoundaries: [...selectedBoundaries, payload],
      };
    },
    removeSelectedBoundary: (state, action: PayloadAction<string>) => {
      return {
        ...state,
        selectedBoundaries: state.selectedBoundaries.filter(
          code => code !== action.payload,
        ),
      };
    },
    clearSelectedBoundaries: state => {
      return { ...state, selectedBoundaries: [] };
    },
    setIsSelectionMode: (
      state,
      { payload: isSelectionMode }: PayloadAction<boolean>,
    ) => {
      const { isSelectionMode: prevIsSelectionMode, ...rest } = state;
      // Same mode do nothing
      if (isSelectionMode === prevIsSelectionMode) {
        return state;
      }
      // If selection mode is turned off, clear selected boundaries
      if (!isSelectionMode) {
        return {
          ...rest,
          isSelectionMode,
          selectedBoundaries: [],
        };
      }
      return {
        ...rest,
        isSelectionMode,
      };
    },
  },
});

// Getters
export const getSelectedBoundaries = (state: RootState) =>
  state.mapSelectionLayerStateSlice.selectedBoundaries;
export const getIsSelectionMode = (state: RootState) =>
  state.mapSelectionLayerStateSlice.isSelectionMode;

// Setters
export const {
  toggleSelectedBoundary,
  clearSelectedBoundaries,
  setIsSelectionMode,
} = mapSelectionLayerStateSlice.actions;

export default mapSelectionLayerStateSlice.reducer;
