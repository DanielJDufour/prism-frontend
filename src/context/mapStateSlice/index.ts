import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { Map as MapBoxMap } from 'mapbox-gl';
import { LayerForm, LayerKey, LayerType } from '../../config/types';
import { LayerData, LayerDataTypes, loadLayerData } from '../layers/layer-data';

interface DateRange {
  startDate?: number;
  endDate?: number; // TODO this field is updated, but doesn't seem to be used yet
}

export type MapState = {
  layers: LayerType[];
  dateRange: DateRange;
  mapboxMap: MapGetter;
  loading: number;
  errors: string[];
  layersData: LayerData<any>[];
  layerForms: LayerForm[];
};

export type FormInputChange = {
  layerId: LayerKey;
  inputId: string;
  value: string;
};

// MapboxGL's map type contains some kind of cyclic dependency that causes an infinite loop in immers's change
// tracking. To save it off, we wrap it in a JS closure so that Redux just checks the function for changes, rather
// than recursively walking the whole object.
type MapGetter = () => MapBoxMap | undefined;

const initialState: MapState = {
  layers: [],
  dateRange: {} as DateRange,
  mapboxMap: (() => {}) as MapGetter,
  // Keep track of loading state with reference counting
  loading: 0,
  errors: [],
  layersData: [],
  layerForms: [],
};

function keepLayer(layer: LayerType, payload: LayerType) {
  // Simple function to control which layers can overlap.
  return payload.type !== layer.type;
}

export const mapStateSlice = createSlice({
  name: 'mapState',
  initialState,
  reducers: {
    addLayer: (
      { layers, layerForms, ...rest },
      { payload }: PayloadAction<LayerType>,
    ) => ({
      ...rest,
      layers: layers.filter(layer => keepLayer(layer, payload)).concat(payload),
      layerForms: layerForms.concat(
        'formInputs' in payload
          ? [{ id: payload.id, inputs: payload.formInputs! }]
          : [],
      ),
    }),

    removeLayer: (
      { layers, layerForms, ...rest },
      { payload }: PayloadAction<LayerType>,
    ) => ({
      ...rest,
      layers: layers.filter(({ id }) => id !== payload.id),
      layerForms: layerForms.filter(({ id }) => id !== payload.id),
    }),

    setFormInputValue: (
      { layerForms, ...rest },
      { payload }: PayloadAction<FormInputChange>,
    ) => ({
      ...rest,
      layerForms: layerForms.map(layerForm =>
        layerForm.id === payload.layerId
          ? {
              id: layerForm.id,
              inputs: layerForm.inputs.map(({ id, value, ...restInput }) => ({
                id,
                value: id === payload.inputId ? payload.value : value,
                ...restInput,
              })),
            }
          : layerForm,
      ),
    }),

    updateDateRange: (state, { payload }: PayloadAction<DateRange>) => ({
      ...state,
      dateRange: payload,
    }),

    setMap: (state, { payload }: PayloadAction<MapGetter>) => ({
      ...state,
      mapboxMap: payload,
    }),

    dismissError: (
      { errors, ...rest },
      { payload }: PayloadAction<string>,
    ) => ({
      ...rest,
      errors: errors.filter(msg => msg !== payload),
    }),
  },
  extraReducers: builder => {
    builder.addCase(
      loadLayerData.fulfilled,
      (
        { layersData, loading, ...rest },
        { payload }: PayloadAction<LayerDataTypes>,
      ) => ({
        ...rest,
        loading: loading - 1,
        layersData: layersData.concat(payload),
      }),
    );

    builder.addCase(
      loadLayerData.rejected,
      ({ loading, errors, ...rest }, action) => ({
        ...rest,
        loading: loading - 1,
        errors: errors.concat(
          action.error.message ? action.error.message : action.error.toString(),
        ),
      }),
    );

    builder.addCase(loadLayerData.pending, ({ loading, ...rest }) => ({
      ...rest,
      loading: loading + 1,
    }));
  },
});

// Setters
export const {
  addLayer,
  removeLayer,
  setFormInputValue,
  updateDateRange,
  setMap,
} = mapStateSlice.actions;

export default mapStateSlice.reducer;
