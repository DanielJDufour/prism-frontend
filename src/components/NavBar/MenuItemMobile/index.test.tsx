import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

import MenuMobile from '.';
import { store } from '../../../context/store';

test('renders as expected', () => {
  const { container } = render(
    <Provider store={store}>
      <MenuMobile />
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
