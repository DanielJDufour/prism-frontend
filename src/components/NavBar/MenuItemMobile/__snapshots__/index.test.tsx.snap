// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <div
    class="MuiPaper-root MuiAccordion-root WithStyles(ForwardRef(Accordion))-root-8 MuiPaper-elevation1"
  >
    <div
      aria-controls="Hazards"
      aria-disabled="false"
      aria-expanded="false"
      class="MuiButtonBase-root MuiAccordionSummary-root WithStyles(ForwardRef(AccordionSummary))-root-10"
      id="Hazards"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content WithStyles(ForwardRef(AccordionSummary))-content-11"
      >
        <img
          alt="Hazards"
          class="MenuMobile-icon-4"
          src="icon_climate.png"
        />
        <mock-typography
          variant="body2"
        >
          Hazards
        </mock-typography>
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-colorInherit MuiIconButton-sizeSmall MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-caret-down fa-w-10 "
            data-icon="caret-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-container MuiCollapse-hidden"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="Hazards"
            id="Hazards"
            role="region"
          >
            <div
              class="MuiAccordionDetails-root WithStyles(ForwardRef(AccordionDetails))-root-13"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
              >
                <div
                  class="MuiGrid-root MenuMobile-categoryContainer-5 MuiGrid-item"
                >
                  <mock-typography
                    classname="MenuMobile-categoryTitle-6"
                    variant="body2"
                  >
                    Indikator Iklim
                  </mock-typography>
                  <hr />
                  <div
                    class="MenuMobile-layersContainer-7"
                  >
                    <mock-switch
                      checked="false"
                      color="default"
                      inputprops="[object Object]"
                      size="small"
                    />
                     
                    <mock-typography
                      variant="body1"
                    >
                      Rainfall Anomaly (Monthly)
                    </mock-typography>
                  </div>
                  <div
                    class="MenuMobile-layersContainer-7"
                  >
                    <mock-switch
                      checked="false"
                      color="default"
                      inputprops="[object Object]"
                      size="small"
                    />
                     
                    <mock-typography
                      variant="body1"
                    >
                      Standardized Precipitation Index (SPI)
                    </mock-typography>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MenuMobile-categoryContainer-5 MuiGrid-item"
                >
                  <mock-typography
                    classname="MenuMobile-categoryTitle-6"
                    variant="body2"
                  >
                    Drought Indicators
                  </mock-typography>
                  <hr />
                  <div
                    class="MenuMobile-layersContainer-7"
                  >
                    <mock-switch
                      checked="false"
                      color="default"
                      inputprops="[object Object]"
                      size="small"
                    />
                     
                    <mock-typography
                      variant="body1"
                    >
                      Agricultural drought (VHI)
                    </mock-typography>
                  </div>
                  <div
                    class="MenuMobile-layersContainer-7"
                  >
                    <mock-switch
                      checked="false"
                      color="default"
                      inputprops="[object Object]"
                      size="small"
                    />
                     
                    <mock-typography
                      variant="body1"
                    >
                      Days Since Last Rain
                    </mock-typography>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MenuMobile-categoryContainer-5 MuiGrid-item"
                >
                  <mock-typography
                    classname="MenuMobile-categoryTitle-6"
                    variant="body2"
                  >
                    Floods Risk
                  </mock-typography>
                  <hr />
                  <div
                    class="MenuMobile-layersContainer-7"
                  >
                    <mock-switch
                      checked="false"
                      color="default"
                      inputprops="[object Object]"
                      size="small"
                    />
                     
                    <mock-typography
                      variant="body1"
                    >
                      Floods Forecast
                    </mock-typography>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root MuiAccordion-root WithStyles(ForwardRef(Accordion))-root-8 MuiPaper-elevation1"
  >
    <div
      aria-controls="Exposure"
      aria-disabled="false"
      aria-expanded="false"
      class="MuiButtonBase-root MuiAccordionSummary-root WithStyles(ForwardRef(AccordionSummary))-root-10"
      id="Exposure"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content WithStyles(ForwardRef(AccordionSummary))-content-11"
      >
        <img
          alt="Exposure"
          class="MenuMobile-icon-4"
          src="icon_basemap.png"
        />
        <mock-typography
          variant="body2"
        >
          Exposure
        </mock-typography>
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-colorInherit MuiIconButton-sizeSmall MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-caret-down fa-w-10 "
            data-icon="caret-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-container MuiCollapse-hidden"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="Exposure"
            id="Exposure"
            role="region"
          >
            <div
              class="MuiAccordionDetails-root WithStyles(ForwardRef(AccordionDetails))-root-13"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
              >
                <div
                  class="MuiGrid-root MenuMobile-categoryContainer-5 MuiGrid-item"
                >
                  <mock-typography
                    classname="MenuMobile-categoryTitle-6"
                    variant="body2"
                  >
                    Demographics
                  </mock-typography>
                  <hr />
                  <div
                    class="MenuMobile-layersContainer-7"
                  >
                    <mock-switch
                      checked="false"
                      color="default"
                      inputprops="[object Object]"
                      size="small"
                    />
                     
                    <mock-typography
                      variant="body1"
                    >
                      Population Density
                    </mock-typography>
                  </div>
                  <div
                    class="MenuMobile-layersContainer-7"
                  >
                    <mock-switch
                      checked="false"
                      color="default"
                      inputprops="[object Object]"
                      size="small"
                    />
                     
                    <mock-typography
                      variant="body1"
                    >
                      Household with Income Derived from Agriculture
                    </mock-typography>
                  </div>
                  <div
                    class="MenuMobile-layersContainer-7"
                  >
                    <mock-switch
                      checked="false"
                      color="default"
                      inputprops="[object Object]"
                      size="small"
                    />
                     
                    <mock-typography
                      variant="body1"
                    >
                      Houshold with Food Expenditure &gt;65%
                    </mock-typography>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root MuiAccordion-root WithStyles(ForwardRef(Accordion))-root-8 MuiPaper-elevation1"
  >
    <div
      aria-controls="Vulnerability"
      aria-disabled="false"
      aria-expanded="false"
      class="MuiButtonBase-root MuiAccordionSummary-root WithStyles(ForwardRef(AccordionSummary))-root-10"
      id="Vulnerability"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content WithStyles(ForwardRef(AccordionSummary))-content-11"
      >
        <img
          alt="Vulnerability"
          class="MenuMobile-icon-4"
          src="icon_vulnerable.png"
        />
        <mock-typography
          variant="body2"
        >
          Vulnerability
        </mock-typography>
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-colorInherit MuiIconButton-sizeSmall MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-caret-down fa-w-10 "
            data-icon="caret-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-container MuiCollapse-hidden"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="Vulnerability"
            id="Vulnerability"
            role="region"
          >
            <div
              class="MuiAccordionDetails-root WithStyles(ForwardRef(AccordionDetails))-root-13"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
              >
                <div
                  class="MuiGrid-root MenuMobile-categoryContainer-5 MuiGrid-item"
                >
                  <mock-typography
                    classname="MenuMobile-categoryTitle-6"
                    variant="body2"
                  >
                    Vulnerable Groups
                  </mock-typography>
                  <hr />
                  <div
                    class="MenuMobile-layersContainer-7"
                  >
                    <mock-switch
                      checked="false"
                      color="default"
                      inputprops="[object Object]"
                      size="small"
                    />
                     
                    <mock-typography
                      variant="body1"
                    >
                      Poverty Rate
                    </mock-typography>
                  </div>
                  <div
                    class="MenuMobile-layersContainer-7"
                  >
                    <mock-switch
                      checked="false"
                      color="default"
                      inputprops="[object Object]"
                      size="small"
                    />
                     
                    <mock-typography
                      variant="body1"
                    >
                      Prevalence of Undernoursihment
                    </mock-typography>
                  </div>
                  <div
                    class="MenuMobile-layersContainer-7"
                  >
                    <mock-switch
                      checked="false"
                      color="default"
                      inputprops="[object Object]"
                      size="small"
                    />
                     
                    <mock-typography
                      variant="body1"
                    >
                      Open Unemployment
                    </mock-typography>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root MuiAccordion-root WithStyles(ForwardRef(Accordion))-root-8 MuiPaper-elevation1"
  >
    <div
      aria-controls="Capacity"
      aria-disabled="false"
      aria-expanded="false"
      class="MuiButtonBase-root MuiAccordionSummary-root WithStyles(ForwardRef(AccordionSummary))-root-10"
      id="Capacity"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content WithStyles(ForwardRef(AccordionSummary))-content-11"
      >
        <img
          alt="Capacity"
          class="MenuMobile-icon-4"
          src="icon_capacity.png"
        />
        <mock-typography
          variant="body2"
        >
          Capacity
        </mock-typography>
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-colorInherit MuiIconButton-sizeSmall MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-caret-down fa-w-10 "
            data-icon="caret-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-container MuiCollapse-hidden"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="Capacity"
            id="Capacity"
            role="region"
          >
            <div
              class="MuiAccordionDetails-root WithStyles(ForwardRef(AccordionDetails))-root-13"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
              >
                <div
                  class="MuiGrid-root MenuMobile-categoryContainer-5 MuiGrid-item"
                >
                  <mock-typography
                    classname="MenuMobile-categoryTitle-6"
                    variant="body2"
                  >
                    Livestock And Crop Preparedness
                  </mock-typography>
                  <hr />
                  <div
                    class="MenuMobile-layersContainer-7"
                  >
                    <mock-switch
                      checked="false"
                      color="default"
                      inputprops="[object Object]"
                      size="small"
                    />
                     
                    <mock-typography
                      variant="body1"
                    >
                      Cereal Yield
                    </mock-typography>
                  </div>
                  <div
                    class="MenuMobile-layersContainer-7"
                  >
                    <mock-switch
                      checked="false"
                      color="default"
                      inputprops="[object Object]"
                      size="small"
                    />
                     
                    <mock-typography
                      variant="body1"
                    >
                      Poor Houshold With Access to Food Assistance
                    </mock-typography>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root MuiAccordion-root WithStyles(ForwardRef(Accordion))-root-8 MuiPaper-elevation1"
  >
    <div
      aria-controls="Tables"
      aria-disabled="false"
      aria-expanded="false"
      class="MuiButtonBase-root MuiAccordionSummary-root WithStyles(ForwardRef(AccordionSummary))-root-10"
      id="Tables"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content WithStyles(ForwardRef(AccordionSummary))-content-11"
      >
        <img
          alt="Tables"
          class="MenuMobile-icon-4"
          src="icon_table.png"
        />
        <mock-typography
          variant="body2"
        >
          Tables
        </mock-typography>
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-colorInherit MuiIconButton-sizeSmall MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-caret-down fa-w-10 "
            data-icon="caret-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-container MuiCollapse-hidden"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="Tables"
            id="Tables"
            role="region"
          >
            <div
              class="MuiAccordionDetails-root WithStyles(ForwardRef(AccordionDetails))-root-13"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
              >
                <div
                  class="MuiGrid-root MenuMobile-categoryContainer-5 MuiGrid-item"
                >
                  <mock-typography
                    classname="MenuMobile-categoryTitle-6"
                    variant="body2"
                  >
                    Socioeconomic Data
                  </mock-typography>
                  <hr />
                  <mock-button
                    classname="MenuMobile-button-2"
                    id="Poverty headcount 2016 vs 2018"
                  >
                    <mock-typography
                      variant="body1"
                    >
                      Poverty headcount 2016 vs 2018
                    </mock-typography>
                  </mock-button>
                </div>
                <div
                  class="MuiGrid-root MenuMobile-categoryContainer-5 MuiGrid-item"
                >
                  <mock-typography
                    classname="MenuMobile-categoryTitle-6"
                    variant="body2"
                  >
                    Price Data
                  </mock-typography>
                  <hr />
                  <mock-button
                    classname="MenuMobile-button-2"
                    id="Monthly hay prices"
                  >
                    <mock-typography
                      variant="body1"
                    >
                      Monthly hay prices
                    </mock-typography>
                  </mock-button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
